scripts/generate-compose.sh

```diff
@@ -0,0 +1,45 @@
+#! /bin/bash
+source .config
+
+compose=`cat ./docker-compose.yml.in`
+
+compose="${compose//REPLACE_WITH_EXTERNAL_HOST/$SETTING_EXTERNAL_HOST}"; 
+compose="${compose//REPLACE_WITH_ADMIN_EMAIL/$SETTING_ZULIP_ADMINISTRATOR}"; 
+
+if [ $POSTGRES_PASSWORD == "REPLACE_WITH_SECURE_POSTGRES_PASSWORD" ]; 
+then 
+    compose="${compose//REPLACE_WITH_SECURE_POSTGRES_PASSWORD/$(tr -dc 'abcdef1234567890' </dev/urandom | head -c 32  ; echo)}"; 
+else 
+    compose="${compose//REPLACE_WITH_SECURE_POSTGRES_PASSWORD/$POSTGRES_PASSWORD}"; 
+fi;
+
+if [ $MEMCACHED_PASSWORD == "REPLACE_WITH_SECURE_MEMCACHED_PASSWORD" ]; 
+then 
+    compose="${compose//REPLACE_WITH_SECURE_MEMCACHED_PASSWORD/$(tr -dc 'abcdef1234567890' </dev/urandom | head -c 32  ; echo)}"; 
+else 
+    compose="${compose//REPLACE_WITH_SECURE_MEMCACHED_PASSWORD/$MEMCACHED_PASSWORD}"; 
+fi;
+
+if [ $RABBITMQ_DEFAULT_PASS == "REPLACE_WITH_SECURE_RABBITMQ_PASSWORD" ]; 
+then 
+    compose="${compose//REPLACE_WITH_SECURE_RABBITMQ_PASSWORD/$(tr -dc 'abcdef1234567890' </dev/urandom | head -c 32  ; echo)}"; 
+else 
+    compose="${compose//REPLACE_WITH_SECURE_RABBITMQ_PASSWORD/$RABBITMQ_DEFAULT_PASS}"; 
+fi;
+
+if [ $REDIS_PASSWORD == "REPLACE_WITH_SECURE_REDIS_PASSWORD" ]; 
+then 
+    compose="${compose//REPLACE_WITH_SECURE_REDIS_PASSWORD/$(tr -dc 'abcdef1234567890' </dev/urandom | head -c 32  ; echo)}"; 
+else 
+    compose="${compose//REPLACE_WITH_SECURE_REDIS_PASSWORD/$REDIS_PASSWORD}"; 
+fi;
+
+if [ $SECRETS_secret_key == "REPLACE_WITH_SECURE_SECRET_KEY" ]; 
+then 
+    compose="${compose//REPLACE_WITH_SECURE_SECRET_KEY/$(tr -dc 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789' </dev/urandom | head -c 64  ; echo)}"; 
+else 
+    compose="${compose//REPLACE_WITH_SECURE_SECRET_KEY/$SECRETS_secret_key}"; 
+fi;
+
+
+echo "$compose" > "docker-compose.yml"
```

Comment:
We should run `shellcheck` on this, and also set it up as a linter for this repository, as it's a super useful tool for avoiding escaping/etc. bugs when doing shell scripting.

(It's already setup as a zulip/zulip linter).

Reply:
That, or we should just write this in Python and give it a test suite; I think that might be a happier approach long-term if we end up adding more features to this.

Reply:
Yup. It would add the expectation the users have python installed but in the long term would be the better option.

Reply:
Shall I go ahead and change it to Python?

Reply:
Yeah, go for it.